### PR TITLE
Snow 1703629 log pip error message

### PR DESCRIPTION
**Pull request by sfc-gh-pczajka**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Add pip error details to logs.
Small refactor of `download_unavailable_packages` (raise exception earlier, as all usages were immediately raising in case `succeded==False`)

### CODE REVIEW

File: src/snowflake/cli/_plugins/snowpark/package_utils.py [hunk 1]
```diff
@@ -222,23 +215,30 @@ def pip_wheel(
...
         )
-    except subprocess.CalledProcessError as e:
-        log.error("Encountered error %s", e.stderr)
-        raise ClickException(f"Encountered error while running pip wheel.")
+        if raise_on_error:
+            raise ClickException(
+                f"pip wheel finished with error code {result.returncode}. Please re-run with --verbose or --debug for more details."
+            )
+    else:
+        log.info("Pip wheel command executed successfully")
```
Comment (sfc-gh-astus) [inline src/snowflake/cli/_plugins/snowpark/package_utils.py:239]: Do we want spam this message?

Comment (sfc-gh-astus) [inline src/snowflake/cli/_plugins/snowpark/package_utils.py:239]: Please decide on one version 😄 `pip wheel` vs `Pip wheel`.

File: tests_integration/test_package.py [hunk 2]
```diff
@@ -231,9 +231,9 @@ def test_package_with_capital_letters(self, directory_for_test, runner):
             for file in files
         )
 
-    @pytest.mark.integration
     @pytest.mark.integration
     def test_incorrect_input(self, runner):
+        # TODO: refactor snowpark, so pip error is always thrown
```
Comment (sfc-gh-astus) [inline tests_integration/test_package.py:236]: For who is this TODO? Do we want to create jira ticket for it?

Comment (sfc-gh-pczajka) [reply tests_integration/test_package.py]: Ah, it was for me from the morning, changes are already applied. Removed :D

File: src/snowflake/cli/_plugins/snowpark/package_utils.py (hunk 1 above)
Comment (sfc-gh-pczajka) [reply src/snowflake/cli/_plugins/snowpark/package_utils.py]: `pip wheel`, fixed

Comment (sfc-gh-pczajka) [reply src/snowflake/cli/_plugins/snowpark/package_utils.py]: I kept that as I liked the log consistency - "running pip" -> "pip error" or "running pip" -> "pip success", no strong opinion though

**Review by sfc-gh-astus — APPROVED** [review]

**Pull request merged**